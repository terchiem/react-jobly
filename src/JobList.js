import React, { useState, useEffect } from 'react';
import JoblyApi from './JoblyApi';
import './JobList.css';

// components
import SearchBar from './SearchBar';
import JobCard from './JobCard';

/** 
 * Displays a list of JobCard components from the jobs 
 * array kept in state. 
 * */

function JobList() {

  const [jobs, setJobs] = useState([]);

  // fetch all jobs from api on initial mount
  useEffect(() => {
    async function loadJobs() {
      const fetchedJobs = await JoblyApi.request('jobs');
      setJobs(fetchedJobs.jobs);
    }
    loadJobs();
  }, []);


  /** Search for jobs through the api. Sets results to state company array */
  async function searchJob(term) {
    const foundJobs = await JoblyApi.request(`jobs?search=${term}`)
    setJobs(foundJobs.jobs);
  }

  /** Creates a CompanyCard for each company object in state */
  function renderJobs() {
    return jobs.map(j => (
      <JobCard
        key={j.id}
        id={j.id}
        title={j.title}
        salary={j.salary}
        equity={j.equity}
        applied={false} // TODO: check against user's jobs
      />
    )); 
  }

  return (
    <div className="JobList">
      <SearchBar search={searchJob} />
      { jobs.length ? renderJobs() : <p>Sorry, no results were found!</p> }
    </div>
  );
}

export default JobList;