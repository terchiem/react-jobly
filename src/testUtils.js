import React from 'react';
import UserContext from './UserContext';

const testUser = {
  username: "test",
  first_name: "test",
  last_name: "test",
  email: "test@email.com",
  photo_url: null
}

const testToken = "test";

function UserProvider({ children, currentUser=testUser, token=testToken }) {
  return (
    <UserContext.Provider value={{currentUser, token}}>
      {children}
    </UserContext.Provider>
  )
};

export { UserProvider };