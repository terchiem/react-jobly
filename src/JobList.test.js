import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import JobList from './JobList';
import { UserProvider } from './testUtils';

test('renders without crashing', () => {
  render(
    <MemoryRouter>
      <UserProvider>
        <JobList />
      </UserProvider>
    </MemoryRouter>
  );
});

test('matches snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <JobList />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});