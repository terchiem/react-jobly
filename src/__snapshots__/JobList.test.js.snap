// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="List"
  >
    <form
      class="SearchBar"
    >
      <input
        class="SearchBar-input"
        placeholder="Search term..."
        value=""
      />
      <button
        class="SearchBar-button"
      >
        Search
      </button>
    </form>
    <div
      class="LoadingSpinner-container"
    >
      <div
        class="LoadingSpinner"
      />
    </div>
  </div>
</DocumentFragment>
`;
